discretisedfield/region.py

```diff
@@ -345,21 +295,25 @@ def units(self):
 
     @units.setter
     def units(self, units):
-        if not isinstance(units, (tuple, list)):
-            raise TypeError(f"units must be of type tuple or list. Not {type(units)}.")
-
-        if not all(isinstance(i, str) for i in units):
-            raise TypeError("units can only contain elements of type str.")
-
-        if len(units) != self.ndim:
-            raise ValueError(
-                "units must have the same length as p1 and p2."
-                f" Not {len(units)=} and {self.ndim=}."
+        if units is None:
+            units = ["m"] * self.ndim
+        elif isinstance(units, (tuple, list)):
+            if not all(isinstance(i, str) for i in units):
+                raise TypeError("units can only contain elements of type str.")
+            if len(units) != self.ndim:
+                raise ValueError(
+                    "units must have the same length as p1 and p2."
+                    f" Not {len(units)=} and {self.ndim=}."
+                )
+        else:
+            raise TypeError(
+                "units must be of type tuple, list, or None (for default behaviour)."
+                f" Not {type(units)}."
             )
 
         self._units = units
 
-    @property
+    @functools.cached_property
```

Comment:
`edges` could change (with scaling) -> no caching
[similar for e.g. center, ... below]

Reply:
Scaling through multiplication and translation will return a new object. Surly as we cannot change pmin and pmax the edges and centre can never change.

Reply:
That is true unless we implement things like `+=` in a way that it directly changes pmin and pmax internally (instead of re-assigning the Region object in the end). Thoughts?

Reply:
And I am still considering adding an optional argument `inplace=True` (defaulting to `False`) to avoid creating a new object if desired.

Reply:
Yeah that is fair enough, they are simple operations so will be a negligible time to compute anyway.